packages/ruby/README.md

```diff
@@ -25,73 +25,71 @@ from the environment, or you may hardcode them.
-### Sensitive Data
+### Payload Data
 
-If you have sensitive data you'd like to prevent from being sent to the Metrics
-API via headers, query params or payload bodies, you can specify a list of keys
-to filter via the `reject_params` option. Key-value pairs matching these keys
-will not be included in the request to the Metrics API.
-
-You are also able to specify a set of `allow_only` which should only be sent through.
-Any header or body values not matching these keys will be filtered out and not
-send to the API.
-
-You may only specify either `reject_params` or `allow_only` keys, not both.
+Option              | Required? | Type             | Description
+--------------------|-----------|------------------|----------
+`api_key`           | yes       | string           | API Key used to make the request. Note that this is different from the `readmeAPIKey` described above in the options data. This should be a value from your API that is unique to each of your users.
+`label`             | no        | string           | This will be the user's display name in the API Metrics Dashboard, since it's much easier to remember a name than an API key.
+`email`             | no        | string           | Email of the user that is making the call.
+`log_id`            | no        | string           | A UUIDv4 identifier. If not provided this will be automatically generated for you. Providing your own `log_id` is useful if you want to know the URL of the log in advance, i.e. `{your_base_url}/logs/{your_log_id}`.
 
 ### Rails
 
 ```ruby
 # config/environments/development.rb or config/environments/production.rb
 require "readme/metrics"
 
-options = {
+sdk_options = {
   api_key: "<<apiKey>>",
   development: false,
   reject_params: ["not_included", "dont_send"],
   buffer_length: 5,
 }
 
-config.middleware.use Readme::Metrics, options do |env|
+config.middleware.use Readme::Metrics, sdk_options do |env|
   current_user = env['warden'].authenticate
 
-  if current_user.present?
-    {
-      api_key: current_user.api_key, # Not the same as the ReadMe API Key
-      label: current_user.name,
-      email: current_user.email
-    }
-  else
-    {
-      api_key: "guest",
-      label: "Guest User",
-      email: "guest@example.com"
-    }
-  end
+  payload_data = current_user.present? ? {
+    api_key: current_user.api_key, # Not the same as the ReadMe API Key
+    label: current_user.name,
+    email: current_user.email
+  } : {
+    api_key: "guest",
+    label: "Guest User",
+    email: "guest@example.com"
+  }
+
+  payload_data
 end
 ```
 
 ### Rack
 
 ```ruby
 # config.ru
-options = {
+sdk_options = {
   api_key: "<<apiKey>>",
   development: false,
   reject_params: ["not_included", "dont_send"]
 }
 
-use Readme::Metrics, options do |env|
+use Readme::Metrics, sdk_options do |env|
     {
       api_key: "owlbert_api_key"
       label: "Owlbert",
-      email: "owlbert@example.com"
+      email: "owlbert@example.com",
+      log_id: SecureRandom.uuid
```

Comment:
Is `SecureRandom` part of the standard Ruby library?

Reply:
Yup, https://ruby-doc.org/stdlib-2.5.1/libdoc/securerandom/rdoc/SecureRandom.html it does need to be required tho